crates/mvr-api/src/metrics/middleware.rs

```diff
@@ -22,10 +22,16 @@ pub(crate) async fn track_metrics(
         .unwrap_or("/UNSUPPORTED")
         .to_string();
 
+    let source_header = req
+        .headers()
+        .get(MVR_SOURCE_HEADER)
+        .map(|h| h.to_str().unwrap_or("unknown_origin").to_string())
+        .unwrap_or("unknown_origin".to_string());
+
```

Comment:
nit: I think it's better to do unwrap_or once 
```suggestion
    let source_header = req
        .headers()
        .get(MVR_SOURCE_HEADER)
        .and_then(|h| h.to_str().ok())
        .unwrap_or("unknown_origin")
        .to_string();
```